csrc/scheduler/matmul_utils.cpp

```diff
@@ -533,16 +552,21 @@ std::shared_ptr<MatmulParams> getMatmulHeuristics(
   params->cparams.index_type = runtime_info.getIndexType();
 
   // Check initial conditions
-  auto mma_exprs = ir_utils::getOpsOfType<MmaOp>(fusion);
-  mma_utils::CombineMulSum combiner(fusion);
+  std::vector<mma_utils::MatmulPattern> patterns =
+      mma_utils::findMatmulPatterns(fusion);
+  NVF_ERROR(!patterns.empty(), "No matmul patterns were found");
   NVF_ERROR(
-      combiner.isValid(),
-      "There's no (single) mma op or mul-sum op which mma op can replace")
+      patterns.size() == 1,
+      "Only a single matmul pattern can currently be fused");
+  mma_utils::MatmulPattern& pattern = patterns.front();
+
+  // IdModel is used to analyze problem shape & layout
+  IdModel id_model(fusion);
```

Comment:
That's a good idea. Currently I build an IdModel in `getMatmulHeuristics` then again in `scheduleMatmul`, and actually I also need to rebuild it after translating `MatmulPattern`s to `MmaOp`s. I think this use case is very similar to the `DomainMap` in the pointwise scheduler that you linked to; we just use `IdModel` instead of `ComputeAtMap`.